specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,318 @@
+
+# Description
+
+This event will be raised before the external protocol launch occurs.
+When an attempt to launch an external protocol is made, the default dialog is displayed in which the user can select `Open` or `Cancel` if the host does not cancel the event.  
+The `NavigationStarting` event will be raised before the `LaunchingExternalProtocol` event, followed by the `NavigationCompleted` event.
+The `SourceChanged`, `ContentLoading`, and `HistoryChanged` events will not be raised when a request is made to launch an external protocol.
+
+The `LaunchingExternalProtocol` event will be raised on the `CoreWebView2` interface.
+
+# Examples
+
+## Win32 C++
+    
+```cpp 
+
+AppWindow* m_appWindow;
+wil::com_ptr<ICoreWebView2> m_webView;
+EventRegistrationToken m_launchingExternalProtocolToken = {};
+
+void RegisterLaunchingExternalProtocolHandler()
+{
+    auto webView16 = m_webView.try_query<ICoreWebView2_16>();
+    if (webView16)
+    {
+        CHECK_FAILURE(webView16->add_LaunchingExternalProtocol(
+            Callback<ICoreWebView2LaunchingExternalProtocolEventHandler>(
+                [this](
+                    ICoreWebView2* sender,
+                    ICoreWebView2LaunchingExternalProtocolEventArgs* args)
+                {
+                    auto showDialog = [this, args]
+                    {
+                        // Set the `Cancel` property to `TRUE`, as we will either silently launch the 
+                        // trusted app or display a custom dialog. 
+                        args->put_Cancel(true);
+                        wil::unique_cotaskmem_string uri;
+                        CHECK_FAILURE(args->get_Uri(&uri));
+                        if (wcsicmp(uri.get(), L"calculator://") == 0)
+                        {
+                            // If this matches our desired protocol, launch the
+                            // calculator app.
+                            std::wstring protocol_url = L"calculator://";
+                            SHELLEXECUTEINFO info = {sizeof(info)};
+                            info.fMask = SEE_MASK_NOASYNC;
+                            info.lpVerb = L"open";
+                            info.lpFile = protocol_url.c_str();
+                            info.nShow = SW_SHOWNORMAL;
+                            ::ShellExecuteEx(&info);
+                        }
+                        else
+                        {
+                            // To display a custom dialog we cancel the launch, display
+                            // a custom dialog, and then manually launch the external protocol. 
+                            wil::unique_cotaskmem_string initiating_origin;
+                            CHECK_FAILURE(args->get_InitiatingOrigin(&initiating_origin));
+                            std::wstring message = L"Launching External Protocol request";
+                            if (initiating_origin.get() == L"")

```

Comment:
This is not how you test a C-style null terminated string for emptiness.

Reply:
Oops. Please fix.